bindings/java/src/org/sleuthkit/datamodel/CommunicationsUtils.java

```diff
@@ -53,22 +53,69 @@ public static String normalizePhoneNum(String phoneNum) throws TskCoreException
 	}
 
 	/**
-	 * Checks if the given string is a valid email address.
 	 * Normalizes the given email address by converting it to lowercase.
 	 *
-	 * @param emailAddress The string to be checked and normalized.
+	 * @param emailAddress The email address string to be normalized.
 	 *
 	 * @return The normalized email address.
 	 * @throws TskCoreException If the given string is not a valid email address.
 	 */
 	public static String normalizeEmailAddress(String emailAddress) throws TskCoreException {
 		
-		EmailValidator validator = EmailValidator.getInstance(true, true);
-        if (validator.isValid(emailAddress)) {
+        if (isValidEmailAddress(emailAddress)) {
             return emailAddress.toLowerCase();
         } else {
             throw new TskCoreException(String.format("Input string is not a valid email address: %s", emailAddress));
         }
 	}
 	
+	/**
+	 * Checks if the given accountId is a valid id for 
+	 * the specified account type.
+	 * 
+	 * @param accountType Account type.
+	 * @param accountUniqueID Id to check.
+	 * 
+	 * @return True, if the id is a valid id for the given account type, False otherwise.
+	 */
+	public static boolean isValidAccountId(Account.Type accountType, String accountUniqueID) {
+		if (accountType == Account.Type.PHONE) {
+			return isValidPhoneNumber(accountUniqueID);
+		}
+		if (accountType == Account.Type.EMAIL) {
+			return isValidPhoneNumber(accountUniqueID);
+		}
+		
+		return !StringUtils.isEmpty(accountUniqueID);
+	}
+	
+	/**
+	 * Checks if the given string is a valid phone number.
+	 *
+	 * @param phoneNum Phone number string to check.
+	 *
+	 * @return True if the given string is a valid phone number, false otherwise.
+	 */
+	public static boolean isValidPhoneNumber(String phoneNum) {
+		if (!StringUtils.isEmpty(phoneNum)) {
+			return phoneNum.matches("\\+?[0-9()\\-\\s]+");
```

Comment:
Where did this validity check come from? Do we need to replace the code where this came from with a call to this method?

Reply:
Not the author, but that is from the normalizePhoneNumber method. I agree that this should call that method and catch the exception to ensure they don't diverge in their validation logic.

Reply:
@raman-bt Could you change this via discussion above?